spec/controllers/groups_controller_spec.rb

```diff
@@ -241,6 +241,66 @@
       end
     end
 
+    describe '#create_groups_when_students_work_alone' do
+      # Create students
+      let!(:students) { create_list(:student, 5) }
+
+      context 'assignment.group_max = 1' do
+        let!(:assignment) { create(:assignment) }
+
+        it 'creates groups for individual students' do
+          data = []
+          students.each do |record|
```

Comment:
Here try using `map` instead of `each` (you may want to look up these methods to understand the difference for creating a new list)